Library/Homebrew/formula_path.sh

```diff
@@ -39,20 +53,35 @@ homebrew-prefix() {
   fi
 
   if [[ -z "${formula_exists}" &&
-        -z "${HOMEBREW_NO_INSTALL_FROM_API}" &&
-        -f "${HOMEBREW_CACHE}/api/formula.json" ]]
+        -z "${HOMEBREW_NO_INSTALL_FROM_API}" ]]
   then
-    formula_exists="$(
-      ruby -rjson <<RUBY 2>/dev/null
-puts 1 if JSON.parse(File.read("${HOMEBREW_CACHE}/api/formula.json")).any? do |f|
-  f["name"] == "${formula}"
-end
-RUBY
-    )"
+    if [[ -f "${HOMEBREW_CACHE}/api/parsed/formula_names.txt" ]] &&
+       grep -Fxq "${formula}" "${HOMEBREW_CACHE}/api/parsed/formula_names.txt"
```

Comment:
```suggestion
       grep --fixed-strings --line-regexp --quiet "${formula}" "${HOMEBREW_CACHE}/api/formula_names.txt"
```

- use longer flag names
- are you confident these all work on all versions of macOS/Linux and behave the same?

Reply:
I think they're POSIX flags but will check.

Reply:
@Bo98 Thanks 🙇🏻

Reply:
These are POSIX flags, but interestingly only in short form: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html